ddtrace/propagation/utils.py

```diff
@@ -6,17 +6,27 @@ def get_wsgi_header(header):
     return "HTTP_{}".format(header.upper().replace("-", "_"))
 
 
+_cached_headers = {}
+
+
 def from_wsgi_header(header):
     """Convert a WSGI compliant HTTP header into the original header.
     See https://www.python.org/dev/peps/pep-3333/#environ-variables for
     information from the spec.
     """
-    HTTP_PREFIX = "HTTP_"
-    # PEP 333 gives two headers which aren't prepended with HTTP_.
-    UNPREFIXED_HEADERS = {"CONTENT_TYPE", "CONTENT_LENGTH"}
+    try:
```

Comment:
this might read better as:

```python
if header in _cached_headers:
    return _cached_headers[header]


# what to do when it isn't there.
```

it'll bring the bulk of this function out of the `except` block

Reply:
I agree, but given that this is a performance enhancement, I thought we could do with EAFP so that we don't lookup the key twice

Reply:
could also do as:

```python
result = _cached_headers.get(header)
if result:
    return result
```

to only do the lookup once.

Reply:
also, depending on the overhead added, it might be worth the performance impact for read/maintainability.